### ISSUE TITLE
(FEATURE) On-the-fly 2D to 3D Video Conversion for VR Headsets 

### ISSUE THREAD

**Issue by threedguru** [UniversalMediaServer/UniversalMediaServer#3486]
Hi,

I posted this as a new feature suggestion:

https://www.universalmediaserver.com/forum/viewtopic.php?f=13&t=15369

mik_s suggested that I come over here to Git hub. I am not sure if this is the right place. I have already developed the solution, but it needs broader proving (e.g. on AMD GPU). It's in my local git repository. I developed it against 11.4.1. 

https://www.universalmediaserver.com/forum/viewtopic.php?f=14&t=15377

I am trying to establish the best way to take it forward.

Any suggestions?

**Comment by SubJunk** [comment]
@threedguru great idea, I would love to see the code, can you submit a pull request so we can see it?

**Comment by threedguru** [comment]
Hi @SubJunk, thanks for replying. I have never done a "pull request"
before, but it sounds like the right next step though. I have just read a
brief overview of how a "pull request" works. I can certainly have a go. I
git cloned the 11.4.1 version of UMS as a starting point. Is there a better
link to follow on "pull requests" than this one?

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests

I have been thinking/working on how to share this with the VR community. To
that extent I have been working on the video linked below. It is a work in
progress from me, but it provides a demo of the solution and contains
screenshots and descriptions of most of the changes that I have made to UMS
so far.

https://streamable.com/sjwdqy

Comments on the video are very much appreciated.

**Comment by threedguru** [comment]
Hi @SubJunk. I have created a pull request as requested. I have created a local fork to perform the pull request. I could not figure out how to make the request against tag 11.4.1 so there are probably some conflicts with main. Anyway, you can see the code. Comments appreciated. The screen shots and my commentary in the video will help to understand what I have done. See video link again below:

https://streamable.com/sjwdqy

**Comment by threedguru** [comment]
Hello again @SubJunk. I have updated the pull request after merging with main, testing and fixing the conflicts. I have provided a more detailed overview in the description of the pull request, a video link and some screen shots. See here:

https://github.com/UniversalMediaServer/UniversalMediaServer/pull/3490

**Comment by threedguru** [comment]
Hi @SubJunk, just for info I have added, to the end of the pull request description, a bit of theoretical background and links to previous work on the topic (on which I have built). I think this was needed to help understand what is going on. I know that this is a bit of an unusual new feature addition and is a little different from a straight small incremental enhancement to the UMS product. I also know that finding time for these types of things is a question of priorities and people's time, it is an open source project after all. Right now, the pull request seems to be stuck in review workflow state. Obviously, I do not have write access for the merge - which the system is also flagging. I am not sure what the next step is. The pull request is here:

https://github.com/UniversalMediaServer/UniversalMediaServer/pull/3490
